### ISSUE TITLE
Support GameNetworkingSockets 1.3.0

### ISSUE THREAD

**Issue by sternenseemann** [opensoldat/opensoldat#59]
Tried updating the library today and it seems like the connection between client and server just times out when connecting to a locally started  server.

I think this has to do with the changes to [locking and threading](https://github.com/ValveSoftware/GameNetworkingSockets/releases/tag/v1.3.0) made in that release, but I am not familiar enough with the soldat codebase to know how to best fix this (error messages say something about likely thread starvation, so seems to check out).

**Comment by BranDougherty** [comment]
Thanks for checking out the new version so quickly. The release notes say that the open source headers are now compatible with the Steamworks SDK headers, but there are still some preprocessor directives in GameNetworkingSockets.pas which are specific to the open source version. I guess they're not required anymore, as when I took them out everything seemed to work. I'll investigate a little more thoroughly then make a PR.

**Comment by BranDougherty** [comment]
I should give an update on this. It wasn't that bad to get v1.3.0 working with 64 bit Linux and Windows, but when I tried it on 32 bit builds it failed. Then I tried v1.2.0 on 32 bit builds and it also failed, so there is some problem with GNS on 32 bit builds. Also, the v1.3.0 release notes say it is compatible with Steamworks 1.51 but the Steam bindings in Soldat are for Steamworks 1.50. We should update both at the same time, but I haven't touched the steam code yet so I'm waiting until I get around to that to update this.
